### DISCUSSION TITLE
Modernize for newer versions of Java

### DISCUSSION THREAD

**Discussion by stratozero** [bcgit/bc-java#1755] [Ideas]
Hi everybody,
probably that's not the right place to put this kind of question, so feel free to redirect me to the right channel if necessary.

I just want to start contributing, and I'm noticing that a large part of the code base can be refactored using features introduced in JDK 8+, eventually coming to sealed interfaces, records, or simply using Java NIO.

So the question is: why there have been separate builds for different versions of java until Java 8, and after that version there's only one artifact and one code base?

Is there any plan to leverage the new features and APIs of the JDK introduced in versions greater than 8?
Finally: how can I contribute to such an improvement?

**Comment by woj-tek** [comment]
> So the question is: why there have been separate builds for different versions of java until Java 8, and after that version there's only one artifact and one code base?

From what I saw BC leverages multi-release jars (see https://www.baeldung.com/java-multi-release-jar)

**Reply by stratozero** [reply]
You're right. I just looked into the root and the core module. Sorry
